src/App.js

```diff
@@ -0,0 +1,78 @@
+import React, { useRef } from "react";
+import { Synth } from "tone";
+import "./App.css";
+
+function App() {
+  let synth = new Synth().toMaster();
+  const inputEl = useRef(null);
+
+  const playTone = (e) => {
+    switch (e.key) {
+      case "d":
+        return synth.triggerAttack("C4");
+      case "r":
+        return synth.triggerAttack("C#4");
+      case "f":
+        return synth.triggerAttack("D4");
+      case "t":
+        return synth.triggerAttack("D#4");
+      case "g":
+        return synth.triggerAttack("E4");
+      case "h":
+        return synth.triggerAttack("F4");
+      case "u":
+        return synth.triggerAttack("F#4");
+      case "j":
+        return synth.triggerAttack("G4");
+      case "i":
+        return synth.triggerAttack("G#4");
+      case "k":
+        return synth.triggerAttack("A4");
+      case "o":
+        return synth.triggerAttack("A#4");
+      case "l":
+        return synth.triggerAttack("B4");
+      default:
+        return;
```

Comment:
instead of using a switch case i would've create another component and  created an array of objects that would look like this:
```javascript 
const data = [
  { id: 'R' ,  sound: 'C4', blackKey: false },
  // the rest of the keys in a similar format
]
```
then map through it and return that other component (named PianoKey) i was talking about like this:
```javascript
{data.map((key, ix) => (
  <PianoKey text={key.id} sound={key.sound} isBlackKey={key.blackKey} />
)}
```
in this other component (PianoKey) you could use the useRef hook to keep track of the keys pressed just as you were doing. Then you can modify your playTone function like this: 
```javascript
const playTone = (e) => {
if(e.keyCode === text.charCodeAt()) {
     // since you're passing into props the sound
     synth.triggerAttack(props.sound);
    }
```